server/tests/unit/test_schema.py

```diff
@@ -0,0 +1,116 @@
+import os
+import sys
+from datetime import datetime
+from django.test import TestCase
+from django.utils import timezone
+from graphene.test import Client
+
+PROJECT_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '../../../')
+)
+
+if PROJECT_PATH not in sys.path:
+    sys.path.append(PROJECT_PATH)
+
+from server.schema import schema
+from server.models import Team, Match, TeamMatch, MLModel, Prediction
+
+
+class TestSchema(TestCase):
+    def setUp(self):
+        self.maxDiff = None
+        self.client = Client(schema)
+
+        home_team = Team(name='Richmond')
+        home_team.save()
+        away_team = Team(name='Melbourne')
+        away_team.save()
+
+        match_datetime = timezone.make_aware(datetime(2018, 5, 5))
+        new_match = Match(start_date_time=match_datetime, round_number=5)
+        new_match.save()
+        match_datetime = timezone.make_aware(datetime(2014, 5, 5))
+        old_match = Match(start_date_time=match_datetime, round_number=7)
+        old_match.save()
+
+        (TeamMatch(team=home_team, match=new_match, at_home=True, score=150)
+         .save())
+        (TeamMatch(team=away_team, match=new_match, at_home=False, score=100)
+         .save())
+        (TeamMatch(team=home_team, match=old_match, at_home=True, score=150)
+         .save())
+        (TeamMatch(team=away_team, match=old_match, at_home=False, score=100)
+         .save())
+
+        ml_model = MLModel(name='test_model')
+        ml_model.save()
+
+        new_prediction = Prediction(
+            match=new_match,
+            ml_model=ml_model,
+            predicted_winner=home_team,
+            predicted_margin=50
+        )
+        new_prediction.save()
+        old_prediction = Prediction(
+            match=old_match,
+            ml_model=ml_model,
+            predicted_winner=away_team,
+            predicted_margin=50
+        )
+        old_prediction.save()
+
+    def assert_correct_results(self, results, expected_results):
+        # graphene returns OrderedDicts instead of dicts, which makes asserting
+        # on results a little more complicated
+        for idx, result in enumerate(results):
+            expected_result = expected_results[idx]
+
+            self.assertEqual(
+                dict(result['match']), expected_result['match'])
+            self.assertEqual(
+                dict(result['mlModel']), expected_result['mlModel'])
+            self.assertEqual(result['isCorrect'], expected_result['isCorrect'])
+
+    def test_predictions(self):
+        expected_predictions = [
+            {
+                'match': {'roundNumber': 5, 'year': 2018},
+                'mlModel': {'name': 'test_model'},
+                'isCorrect': True
+            },
+            {
+                'match': {'roundNumber': 7, 'year': 2014},
```

Comment:
is this roundNumber camel case for some particular reason, or "just because and ain't gonna change it" ?

Reply:
GraphQL magically converts camelcase from the frontend to snakecase in the backend, then converts it back to camelcase in the JSON returned to the frontend. So, when matching attributes of the JSON returned by GraphQL, I need to use camelcase.